app/services/distribution_creator.rb

```diff
@@ -0,0 +1,16 @@
+class DistributionCreator
```

Comment:
One of the biggest advantages to adding service objects like this is that you can unit test their logic. Should we add tests in `spec/services/distribution_creator_spec.rb` for this?

Reply:
Totally! I'll spin it out to another PR though - we already have very thorough coverage here.

Reply:
I was thinking about the naming. How about instead of `DistributionCreator` we use `DistributionCreationService` or `DistributionCreateService`?